commercial/app/model/commercial/events/Eventbrite.scala

```diff
@@ -0,0 +1,187 @@
+
+    def buildEventWithImageSrc(event: EBEvent, src: String) = {
+      new EBEvent(
+        event.id,
+        event.name,
+        event.startDate,
+        event.url,
+        event.description,
+        Some(src),
+        event.status,
+        event.venue,
+        event.tickets,
+        event.capacity
+      )
+    }
+  }
+
+  case class EBTicket(price: Double, quantityTotal: Int, quantitySold: Int)
+
+  object EBTicket {
+
+    def buildTicket(hidden: Boolean, donation: Boolean, valuePence: Double, quantityTotal: Int, quantitySold: Int): Option[EBTicket] = {
+      if (hidden || donation) {
+        None
+      } else {
+        Some(EBTicket(valuePence / 100, quantityTotal, quantitySold))
+      }
+    }
+
+    implicit val ticketReads: Reads[Option[EBTicket]] = (
+      (JsPath \ "hidden").read[Boolean] and
+        (JsPath \ "donation").read[Boolean] and
+        (JsPath \ "cost" \ "value").read[Double].orElse(Reads.pure(0.00)) and
+        (JsPath \ "quantity_total").read[Int] and
+        (JsPath \ "quantity_sold").read[Int]
+      ) (EBTicket.buildTicket _)
+  }
+
+  case class EBVenue(name: Option[String],
+                     address: Option[String],
+                     address2: Option[String],
+                     city: Option[String],
+                     country: Option[String],
+                     postcode: Option[String]) {
+
+    val description = Seq(name, city orElse country).flatten mkString ", "
+  }
+
+  object EBVenue {
+
+    implicit val venueReads: Reads[EBVenue] = {
+
+      def captureEmptyString(x: Reads[Option[String]]): Reads[Option[String]] = {
+        x map (el => if (el.getOrElse("").length == 0) None else el)
+      }
+
+      (
+        captureEmptyString((JsPath \ "name").readNullable[String]) and
+          captureEmptyString((JsPath \ "address" \ "address_1").readNullable[String]) and
+          captureEmptyString((JsPath \ "address" \ "address_2").readNullable[String]) and
+          captureEmptyString((JsPath \ "address" \ "city").readNullable[String]) and
+          captureEmptyString((JsPath \ "address" \ "country").readNullable[String]) and
+          captureEmptyString((JsPath \ "address" \ "postal_code").readNullable[String])
+        ) (EBVenue.apply _)
+    }
+  }
+
+    object Helper {
+
+      def parsePageOfEvents(json: JsValue): Seq[EBEvent] = {
+
+        val response: EBResponse = json.as[EBResponse]
+        response.events.filter(_.startDate.isAfter(now.plusWeeks(2)))
+      }
+
+      def parsePagesOfEvents(feedMetaData: FeedMetaData, feedContent: => Option[String]): Future[ParsedFeed[EBEvent]] = {
+
+        feedMetaData.switch.isGuaranteedSwitchedOn flatMap { switchedOn =>
+          if (switchedOn) {
+            val start = currentTimeMillis
+            feedContent map { body =>
+              val JsArray(pages) = Json.parse(body).as[JsArray]
```

Comment:
This could be done in one go as
`val responses = Json.parse(body).as[Seq[EBResponse]]`